analyzers/src/SonarAnalyzer.CSharp/Rules/BooleanLiteralUnnecessary.cs

```diff
@@ -50,8 +52,43 @@ protected override void Initialize(SonarAnalysisContext context)
             context.RegisterNodeAction(CheckNotEquals, SyntaxKind.NotEqualsExpression);
             context.RegisterNodeAction(CheckConditional, SyntaxKind.ConditionalExpression);
             context.RegisterNodeAction(CheckForLoopCondition, SyntaxKind.ForStatement);
+            context.RegisterNodeAction(CheckIsPatternExpression, SyntaxKindEx.IsPatternExpression);
         }
 
+        private void CheckIsPatternExpression(SonarSyntaxNodeReportingContext context)
+        {
+            if (!IsPatternExpressionSyntaxWrapper.IsInstance(context.Node)
+                || !ConstantPatternSyntaxWrapper.IsInstance(((IsPatternExpressionSyntaxWrapper)context.Node).Pattern) // Temporary to avoid "is not"
+                || IsInsideTernaryWithThrowExpression(context.Node)
+                || CheckForNullabilityAndBooleanConstantsReport(context, (IsPatternExpressionSyntaxWrapper)context.Node, reportOnTrue: true))
+            {
+                return;
+            }
+
+            var isPatternWrapper = (IsPatternExpressionSyntaxWrapper)context.Node;
+
+            CheckForBooleanConstantOnLeft(context, isPatternWrapper, IsTrueLiteralKind, ErrorLocation.BoolLiteralAndOperator);
+            CheckForBooleanConstantOnLeft(context, isPatternWrapper, IsFalseLiteralKind, ErrorLocation.BoolLiteral);
+
+            CheckForBooleanConstantOnRight(context, isPatternWrapper, IsTrueLiteralKind, ErrorLocation.BoolLiteralAndOperator);
+            CheckForBooleanConstantOnRight(context, isPatternWrapper, IsFalseLiteralKind, ErrorLocation.BoolLiteral);
```

Comment:
These 4 lines are a duplication of the base class "CheckEquals" method. Reuse the base class logic instead.